src/compute_surf.cpp

```diff
@@ -437,34 +444,45 @@ int ComputeSurf::tallyinfo(surfint *&ptr)
 }
 
 /* ----------------------------------------------------------------------
-   sum tally values to per-surf values
-   index = 0 if this compute produces a vector
-   index = 1 to N if this compute produces an array with 1 to N columns
-   regardless, result is stored in vector_surf for Nown surfs this proc owns
+   sum tally values to owning surfs via surf->collate()
 ------------------------------------------------------------------------- */
 
-void ComputeSurf::tallysum(int index)
+void ComputeSurf::post_process_surf()
 {
-  // NOTE: could have a way to skip indices that have already been
-  //       tallysum() on this timestep
-  //       would require storing multiple vector_surfs, or in array_surfs
+  if (combined) return;
+  combined = 1;
 
-  //  if (update->ntimestep == last_tallysum) return;
-  last_tallysum = update->ntimestep;
+  // reallocate array_surf if necessary
 
-  // allocate vector surf if necessary
+  int nown = surf->nown;
 
-  if (!vector_surf) memory->create(vector_surf,surf->nown,"surf:vector_surf");
-  
-  // array_surf_tally can be NULL if this proc has performed no tallies
+  if (nown > maxsurf) {
+    memory->destroy(array_surf);
+    maxsurf = nown;
+    memory->create(array_surf,maxsurf,ntotal,"surf:array_surf");
+  }
+
+  // zero array_surf
+
+  int i,j;
+  for (i = 0; i < nown; i++)
+    for (j = 0; j < ntotal; j++)
+      array_surf[i][j] = 0.0;
 
+  // collate entire array of results
+
+  surf->collate_array(ntally,ntotal,tally2surf,array_surf_tally,array_surf);
+
+  /*
   if (array_surf_tally)
     surf->collate_vector(ntally,tally2surf,
                          &array_surf_tally[0][index-1],ntotal,vector_surf);
   else 
     surf->collate_vector(ntally,tally2surf,NULL,ntotal,vector_surf);
+  */
```

Comment:
Can we remove this commented out code?

Reply:
yes, you can remove the printf() statement and revert to
just the error->one() call.

Reply:
that comment was for the collide.cpp.  Just leave the commented out code
in compute_surf.cpp, in case we ever want to include collating one vector
at a time.